src/cow-shed/CoWShedHooks.ts

```diff
@@ -0,0 +1,123 @@
+import {
+  arrayify,
+  defaultAbiCoder,
+  getCreate2Address,
+  joinSignature,
+  solidityKeccak256,
+  splitSignature,
+} from 'ethers/lib/utils'
+import { COW_SHED_FACTORY, COW_SHED_IMPLEMENTATION, SupportedChainId } from 'src/common'
+import { COW_SHED_712_TYPES, ICoWShedCall, ICoWShedOptions } from './types'
+import { COW_SHED_PROXY_INIT_CODE } from './proxyInitCode'
+import type { Signer } from '@ethersproject/abstract-signer'
+import { getCoWShedFactoryInterface } from './contracts'
+import { TypedDataDomain } from 'ethers'
+import {
+  EcdsaSigningScheme,
+  hashTypedData,
+  isTypedDataSigner,
+  SigningScheme,
+  TypedDataTypes,
+} from '@cowprotocol/contracts'
+
+export class CowShedHooks {
+  constructor(private chainId: SupportedChainId, private customOptions?: ICoWShedOptions) {}
+
+  computeProxyAddress(user: string) {
```

Comment:
I would preffer using `COWShedFactory.proxyOf()` view call instead, since the function is already implemented on the smart-contract side